### PR TITLE
Bug 2087981: Change "create" sequence with powering on the vm after clone

### PR DESCRIPTION
**Pull request by lobziik**
Stop passing PowerOn flag and explicitly power on the vm after
successful clone.

If PowerOn flag set to true, vCenter creates additional power on task
after the clone, which we do not know about. Such nuance leads to
possibility of stucking machine in "Provisioned" state
in case if "power on" task did not finish successfully.
This pr adds explicit powering on the machine with saving task reference
and respective checks later. For achieve vm powering on vm separately, during create procedure - exists method also be altered, if machine powered off and sits in "provisioning" phase, it treats as non existed yet.

Additionally powerOn function tries to use datacenter's method "PowerOnMultiVM"
which respects vSphere DRS. If datacenter not provided, function will fallback
to powering on via old PowerOnVM method.

### CODE REVIEW

**Comment by jcpowermac** [comment]
Hmm, what if DRS set the manual?

I _think_ you would need to override automation level.
https://github.com/vmware/vic/blob/6c70cfedabb689d0b97bfa32ba3bc92119ec4860/pkg/vsphere/vm/vm.go#L966-L1004

**Comment by lobziik** [comment]
@jcpowermac, good point, thanks.

Just question, wouldn't it be surprising and undesirable to our users if we would override this parameter?

From the other side, ensure that automation in place makes sense to me, given that autoscaler or some another automation might be around.

**Comment by lobziik** [comment]
I would like to have a look on vsphere e2e results first, then gonna push another commit which overrides DRS automation level.

**Comment by jcpowermac** [comment]
> @jcpowermac, good point, thanks.
> 
> Just question, wouldn't it be surprising and undesirable to our users if we would override this parameter?

If the override isn't there and the cluster is set to manual an admin would have to manually approve the power on location.

> 
> From the other side, ensure that automation in place makes sense to me, given that autoscaler or some another automation might be around.

**Review by elmiko — COMMENTED** [review]
this generally makes sense to me, even the "crutch" ;)

i am curious though, should we have a section to detect if the instance is in suspended state but somehow the machine is in provisioning phase? (i'm wondering if we should have an error log for users who want to mess around with the vCenter or something)

**Comment by lobziik** [comment]
>i am curious though, should we have a section to detect if the instance is in suspended state but somehow the machine is in provisioning phase? (i'm wondering if we should have an error log for users who want to mess around with the vCenter or something)

I thought through such scenario - it might happen, if user stopped vm and manually put machine into "provisioning" phase. In such situation, vm will be started by machine-controller and go to "provisioned" phase during next reconciliation. Or, if task would be not found, it will stuck (powering it on the vm then will help).

This problem there is way more deepen actually, i think we should consider to extend vsphere-specific status api, and track vcenter tasks more precisely.

**Comment by elmiko** [comment]
thanks. that makes sense, and i agree that it would be an extreme case for the user to create this scenario.

File: pkg/controller/vsphere/util.go [hunk 1]
```diff
@@ -181,6 +181,13 @@ func getInsecureFlagFromConfig(config *vSphereConfig) bool {
 	return false
 }
 
+func stringPointerDeref(stringPointer *string) string {
```
Comment (JoelSpeed) [inline pkg/controller/vsphere/util.go:184]: Could you use `pointer.StringDeref` instead? `k8s.io/utils/pointer`

**Review by JoelSpeed — COMMENTED** [review]
Looks good, one nit, WDYT?

**Review by elmiko — COMMENTED** [review]
/lgtm

**Pull request merged**